Cargo.toml

```diff
@@ -26,3 +33,18 @@ sha2 = { version = "0.10", default-features = false }
 # bdk_chain = { path = "../bdk/crates/chain" }
 bdk_chain = { git = "https://github.com/LLFourn/bdk.git", rev = "165f06b8f78bd24f57dfbfd66da1faa189c67457" }
 bdk_electrum = { git = "https://github.com/LLFourn/bdk.git", rev = "165f06b8f78bd24f57dfbfd66da1faa189c67457" }
+
+[profile.dev]
+# Rust debug is too slow.
+# For debug builds always builds with some optimization
+opt-level = "s"
+
+[profile.release]
+# LLVM can perform better optimizations using a single thread
+codegen-units = 1
+debug = 2
+debug-assertions = false
+incremental = false
+lto = 'fat'
+opt-level = 's'
```

Comment:
Why are we doing this in the workspace Cargo.toml and not the individual crates. Surely we'll want a different profile for device than everything else.

Reply:
Cargo complains profiles should be in the root Cargo.toml

Reply:
Apparently not possible so I guess we'll just suck it up for now but at some point we may want to split out device into its own repo.